CONTRIBUTING.MD

```diff
@@ -0,0 +1,105 @@
+
+For those wishing to contribute we highly suggest reading the [dbt-core](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md). Almost all of the information there is applicable to contributing here, too!
+
+### Signing the CLA
+
+Please note that all contributors to `dbt-snowflake` must sign the [Contributor License Agreement](https://docs.getdbt.com/docs/contributor-license-agreements) to have their Pull Request merged into an `dbt-snowflake` codebase. If you are unable to sign the CLA, then the `dbt-snowflake` maintainers will unfortunately be unable to merge your Pull Request. You are, however, welcome to open issues and comment on existing ones.
+
+
+## Getting the code 
+
+You will need `git` in order to download and modify the `dbt-snowflake` source code. On macOS, the best way to download git is to just install [Xcode](https://developer.apple.com/support/xcode/).
+
+### External contributors
+
+If you are not a member of the `dbt-labs` GitHub organization, you can contribute to `dbt-snowflake` by forking the `dbt-snowflake` repository. For a detailed overview on forking, check out the [GitHub docs on forking](https://help.github.com/en/articles/fork-a-repo). In short, you will need to:
+
+1. fork the `dbt-snowflake` repository
+2. clone your fork locally
+3. check out a new branch for your proposed changes
+4. push changes to your fork
+5. open a pull request against `dbt-labs/dbt-snowflake` from your forked repository
+
+### dbt Labs contributors
+
+If you are a member of the `dbt Labs` GitHub organization, you will have push access to the `dbt-snowflake` repo. Rather than forking `dbt-snowflake` to make your changes, just clone the repository, check out a new branch, and push directly to that branch.
+
+
+## Running `dbt-snowflake` in development
+
+### Installation
+
+First make sure that you set up your `virtualenv` as described in [Setting up an environment](https://github.com/dbt-labs/dbt-core/blob/HEAD/CONTRIBUTING.md#setting-up-an-environment).  Ensure you have the latest version of `pip` installed with `pip install --upgrade pip`. Next, install `dbt-snowflake` latest dependencies:
+
+```sh
+pip install -e . -r dev-requirements.txt
+```
+
+When `dbt-snowflake` is installed this way, any changes you make to the `dbt-snowflake` source code will be reflected immediately in your next `dbt-snowflake` run.
+
+To confirm you have correct version of `dbt-core` installed please run `dbt --version` and `which dbt`.
+
+
+## Testing
+
+### Initial Setup
+
+`dbt-snowflake` contains [unit](https://github.com/dbt-labs/dbt-snowflake/tree/main/tests/unit) and [integration](https://github.com/dbt-labs/dbt-snowflake/tree/main/tests/integration) tests. Integration tests require testing against an actual Snowflake warehouse. We have CI set up to test against a Snowflake warehouse during PR checks. 
```

Comment:
```suggestion
`dbt-snowflake` contains [unit](https://github.com/dbt-labs/dbt-snowflake/tree/main/tests/unit) and [integration](https://github.com/dbt-labs/dbt-snowflake/tree/main/tests/integration) tests. Integration tests require testing against an actual Snowflake warehouse. There are two main options for running tests:

- This repository has a CI pipeline active. It runs tests against a Snowflake warehouse during PR checks. 
- You may also run integration tests locally. This requires a `test.env` file in the root of your snowflake repository. This file must contain credentials for accessing a valid Snowflake account.
```
Figured it might be beneficial to more clearly group and format testing methods.